src/onnx/parse_qlinearadd.cpp

```diff
@@ -0,0 +1,202 @@
+{
+    std::vector<op_desc> operators() const { return {{"QLinearAdd"}}; }
+
+    // basic type checking for QLinearAdd Operator
+    void check_inputs(const std::vector<instruction_ref>& args) const
+    {
+        if(args.size() < 7)
+            MIGRAPHX_THROW("QLINEARADD: missing inputs");
+
+        const auto& in_a = args[0];
+        const auto& in_b = args[3];
+
+        auto sh_a = in_a->get_shape();
+        auto sh_b = in_b->get_shape();
+        if(sh_a != sh_b)
+            MIGRAPHX_THROW("QLINEARADD: mismatched input shapes");
+
+        auto type_a = sh_a.type();
+        auto type_b = sh_b.type();
+        if(type_a != migraphx::shape::int8_type and type_a != migraphx::shape::uint8_type)
+            MIGRAPHX_THROW("QLINEARADD: unsupported input type");
+        if(type_b != migraphx::shape::int8_type and type_b != migraphx::shape::uint8_type)
+            MIGRAPHX_THROW("QLINEARADD: unsupported input type");
+    }
+
+    instruction_ref bcast_scalar_instr(const migraphx::shape& shape_out,
+                                       const instruction_ref& arg_in,
+                                       const onnx_parser::node_info& info) const
+    {
+        auto bcast_instr_out = info.add_instruction(
+            migraphx::make_op("multibroadcast", {{"out_lens", shape_out.lens()}}), arg_in);
+        return bcast_instr_out;
+    }
+
+    // This method is to prep for quantizelinear or dequantizelinear operation for
+    // either the broadcasting of weight-scale or zero-points of qlinearadd operator
+    // outputs: operator op (inputs x, broadcasted: scale (float) & zero_pt (8-bit))
+    instruction_ref bcast_qdq_instr(const std::string& op_name,
+                                    const instruction_ref& x_in,
+                                    const instruction_ref& arg_fscale,
+                                    const instruction_ref& arg_z_pt,
+                                    const onnx_parser::node_info& info) const
+    {
+        auto in_lens = x_in->get_shape().lens();
+
+        // prep 1: broadcast scale. it can come as a scalar or a 1-D tensor.
+        std::vector<float> sc_val;
+        auto ev_arg_fscale = arg_fscale->eval();
+        ev_arg_fscale.visit([&](auto s) { sc_val.assign(s.begin(), s.end()); });
+        shape sh_scale = {shape::float_type, {sc_val.size()}};
+        instruction_ref bcast_scale;
+        if(sc_val.size() > 1)
+            bcast_scale = info.add_instruction(
+                migraphx::make_op("broadcast", {{"axis", 0}, {"out_lens", in_lens}}),
+                info.add_literal(sh_scale, sc_val));
+        else
+            bcast_scale =
+                info.add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", in_lens}}),
+                                     info.add_literal(sh_scale, sc_val));
+
+        // prep 2: broadcast zero point. it can come as a scalar or a 1-D tensor.
+        std::vector<int> z_pt_val;
+        auto ev_arg_z_pt = arg_z_pt->eval();
```

Comment:
Same here. This is only used to check the size, so it would be better to use `elements()` instead.

Reply:
Already fixed.